axiom/tests/test_span_event.c

```diff
@@ -256,6 +256,47 @@ static void test_span_event_category(void) {
   tlib_pass_if_str_equal("Category should be the one we set - http", "http",
                          nr_span_event_get_category(event));
 
+  nr_span_event_set_category(event, NR_SPAN_MESSAGE);
+  tlib_pass_if_str_equal("Category should be the one we set - message",
+                         "message", nr_span_event_get_category(event));
+
+  nr_span_event_destroy(&event);
+}
+
+static void test_span_event_spankind(void) {
+  nr_span_event_t* event = nr_span_event_create();
+
+  // Test : the default is NULL (spankind must be explicitly set)
+  tlib_pass_if_str_equal("The default category", NULL,
+                         nr_span_event_get_spankind(event));
+
+  // Test : A null event returns NULL
+  tlib_pass_if_null("The default category", nr_span_event_get_spankind(NULL));
+
+  // Test : passing a NULL event should not blow up
+  nr_span_event_set_spankind(NULL, NR_SPAN_PRODUCER);
+
+  // Test : setting the spankind back and forth
+  nr_span_event_set_spankind(event, NR_SPAN_NO_SPANKIND);
+  tlib_pass_if_str_equal(
+      "Spankind should be the one we set - no spankind (NULL)", NULL,
+      nr_span_event_get_spankind(event));
+
+  // Test : setting the spankind back and forth
+  nr_span_event_set_spankind(event, NR_SPAN_PRODUCER);
+  tlib_pass_if_str_equal("Spankind should be the one we set - producer",
+                         "producer", nr_span_event_get_spankind(event));
+
+  // Test : setting the spankind back and forth
+  nr_span_event_set_spankind(event, NR_SPAN_CLIENT);
+  tlib_pass_if_str_equal("Spankind should be the one we set - client", "client",
+                         nr_span_event_get_spankind(event));
+
+  // Test : setting the spankind back and forth
+  nr_span_event_set_spankind(event, NR_SPAN_CONSUMER);
+  tlib_pass_if_str_equal("Spankind should be the one we set - consumer",
+                         "consumer", nr_span_event_get_spankind(event));
+
```

Comment:
Do we care to test with invalid input, i.e. `spankind` argument passed is out of valid  range like this: `nr_span_event_set_spankind(event, 255);`?

Reply:
Something like this:
```suggestion
  nr_span_event_set_spankind(event, 255);
  tlib_pass_if_str_equal("Invalid spankind sets spankind to none (NULL)",
                         NULL, nr_span_event_get_spankind(event));

```

Reply:
Please see: https://github.com/newrelic/newrelic-php-agent/pull/991/commits/862604b081029206c2072275588076691d30560a